dingo/core/utils/plotting.py

```diff
@@ -0,0 +1,108 @@
+from itertools import zip_longest
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import corner
+import numpy as np
+import pandas as pd
+
+
+def plot_corner_multi(
+    samples, weights=None, labels=None, filename="corner.pdf", **kwargs
+):
+    """
+    Generate a corner plot for multiple posteriors.
+
+    Parameters
+    ----------
+    samples : list[pd.DataFrame]
+        List of sample sets. The DataFrame column names are used as parameter labels.
+    weights : list[np.ndarray or None] or None
+        List of weights sets. The length of each array should be the same as the length of
+        the corresponding samples.
+    labels : list[str or None] or None
+        Labels for the posteriors.
+    filename : str
+        Where to save samples.
+    **kwargs :
+        Forwarded to corner.corner.
+    """
+    # Define plot properties
+    cmap = "Dark2"
+    corner_params = {
+        "smooth": 1.0,
+        "smooth1d": 1.0,
+        "plot_datapoints": False,
+        "plot_density": False,
+        "plot_contours": True,
+        "levels": [0.5, 0.9],
+        "bins": 30,
+    }
+    corner_params.update(kwargs)
+
+    serif_old = mpl.rcParams["font.family"]
+    mpl.rcParams["font.family"] = "serif"
+    linewidth_old = mpl.rcParams["lines.linewidth"]
+    mpl.rcParams["lines.linewidth"] = 2.5
+
+    # In case a single corner plot is desired, convert to lists to iterate.
+    if not isinstance(samples, list):
+        samples = [samples]
+    if not isinstance(weights, list):
+        weights = [weights]
+    if not isinstance(labels, list):
+        labels = [labels]
+
+    fig = None
+    handles = []
+    for i, (s, w, l) in enumerate(zip_longest(samples, weights, labels)):
+        color = mpl.colors.rgb2hex(plt.get_cmap(cmap)(i))
+        fig = corner.corner(
+            s.to_numpy(),
+            labels=s.columns,
+            weights=w,
+            color=color,
+            no_fill_contours=True,
+            fig=fig,
+            **corner_params
+        )
+        handles.append(
+            plt.Line2D([], [], color=color, label=l, linewidth=5, markersize=20)
+        )
+
+    # Eliminate spacing between the 2D plots
+    fig.subplots_adjust(wspace=0, hspace=0)
```

Comment:
I guess this is up to the user, should it be 0? Sometimes this makes the posteriors a bit hard to interpret I attach a photo here; 
![image](https://user-images.githubusercontent.com/40392612/228257525-28e89056-d054-43e4-9148-91ce0f3c93f1.png)

Vs when we were using chainconumser: 
![image](https://user-images.githubusercontent.com/40392612/228257703-b161da12-e525-4c5e-a575-8eebdaef61a2.png)

Reply:
I think ChainConsumer reduces the separation to 0 when we have more than a certain number of parameters. I just set this for all cases, but if you want to base it on `len(common_parameters)` feel free to do so.